proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/logging/ProxyNodeLoggingFilter.java

```diff
@@ -29,25 +30,24 @@
 
     @Override
     public void filter(ContainerRequestContext requestContext) {
-        String istioTraceId = requestContext.getHeaderString(X_B3_TRACEID);
-        if (istioTraceId != null && !istioTraceId.isBlank()) {
-            MDC.put(X_B3_TRACEID, istioTraceId);
-        }
+        ofNullable(requestContext.getHeaderString(X_B3_TRACEID))
+            .filter(not(String::isBlank)).ifPresent(traceId -> MDC.put(X_B3_TRACEID, traceId));
 
         ProxyNodeLogger.addContext(PROXY_NODE_JOURNEY_ID, getJourneyId(requestContext));
-        ProxyNodeLogger.addContext(REFERER, Optional.ofNullable(requestContext.getHeaderString(HttpHeaders.REFERER)).orElse(""));
-        Optional.ofNullable(requestContext.getUriInfo()).ifPresent(u -> ProxyNodeLogger.addContext(RESOURCE_PATH, u.getAbsolutePath().toString()));
```

Comment:
We should also keep the `Optional` in `Optional.ofNullable` as in this case it's less clear without the full name